hermeto/core/package_managers/bundler/gem_models.py

```
@@ -0,0 +1,191 @@
+    """
+
+    root: RootedPath
+    subpath: str
+
+    @pydantic.model_validator(mode="after")
+    def validate_subpath(self) -> Self:
+        """Validate that the subpath is within the package root."""
+        try:
+            self.root.join_within_root(self.subpath)
+        except PathOutsideRoot:
+            raise ValueError("PATH dependencies should be within the package root")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)